src/components/CustomCollection/CustomBookmark/CustomBookmarkForm.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect } from 'react'
+import { useFormik } from 'formik'
+import * as yup from 'yup'
+import {
+  Form,
+  ButtonGroup,
+  Button,
+  Label,
+  TextInput,
+  Alert,
+  ErrorMessage,
+} from '@trussworks/react-uswds'
+
+type CustomBookmarkFormProps = {
+  onSave: (url: string, label: string) => void
+  onCancel: () => void
+  onDelete?: () => void
+  label?: string
+  url?: string
+  showAddWarning?: boolean
+  nameInputRef: React.RefObject<HTMLInputElement>
+  urlInputRef: React.RefObject<HTMLInputElement>
+}
+export const CustomBookmarkForm = ({
+  onSave,
+  onCancel,
+  onDelete,
+  label,
+  url,
+  showAddWarning,
+  nameInputRef,
+  urlInputRef,
+}: CustomBookmarkFormProps) => {
+  // #TODO: Integrate Formik into our forms following the
+  // wrapper component pattern used in other Truss projects
+
+  const formik = useFormik({
+    initialValues: {
+      // We have to provide label and url as possible initial values
+      // in case a user is editing
+      bookmarkLabel: label || '',
+      bookmarkUrl: url || '',
+    },
+
+    validateOnChange: true,
+    validateOnBlur: true,
+    enableReinitialize: true,
+    onSubmit: () => {
+      onSave(formik.values.bookmarkUrl, formik.values.bookmarkLabel)
+      formik.resetForm()
+    },
+
+    validationSchema: yup.object({
+      bookmarkLabel: yup.string().required('Link name is required'),
+      bookmarkUrl: yup
+        .string()
+        .transform((_, value) => {
+          // If the user enters a URL without a scheme,
+          // add http:// so we can validate it.
+          // This doesn't affect the value of the input.
+          if (!value.startsWith('http')) {
```

Comment:
Thought(non-blocking): this is the third time I've seen this check / modification. I'm wondering if it'd be possible to make a small util function to reuse so we don't have to hunt for it if we ever have to adjust what schemes we support?

Reply:
i like this! will see where a good place might be for it to live and run it by yall in slack